homeassistant/components/tailwind/coordinator.py

```diff
@@ -0,0 +1,40 @@
+"""Data update coordinator for Tailwind."""
+from datetime import timedelta
+
+from gotailwind import Tailwind, TailwindDeviceStatus, TailwindError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER
+
+
+class TailwindDataUpdateCoordinator(DataUpdateCoordinator[TailwindDeviceStatus]):
+    """Class to manage fetching Tailwind data."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
```

Comment:
And can we maybe pass the Tailwind as parameter and do a status check in `__init__.py` so we know if the device is available?

Reply:
Why?

Reply:
You could catch `TailwindConnectionError` and raise `ConfigEntryNotReady`

Reply:
That isn't needed, the coordinator already does that. That is a specifically built behavior that happens on the first refresh.

Reply:
I did not know